Conditionally display general date field on author profile

seabelis opened the pull request:
<!-- What issue does this PR close? -->
Closes #3421 

<!-- What does this PR achieve? [feature|hotfix|fix|refactor] -->
Fixes uneditable date on author records.
### Technical
<!-- What should be noted about the implementation? -->

### Testing
<!-- Steps for reviewer to reproduce/verify what this PR does/fixes. -->
Go to http://localhost:8080/authors/OL1095882A/Cid
Add birth or death date.
Save.
See if general date is displayed (should not be).
ALSO
Try to edit/remove general date field. Should no longer appear on profile once saved.
### Screenshot
<!-- If this PR touches UI, please post evidence (screenshots) of it behaving correctly. -->
<img width="952" alt="Screenshot 2020-12-13 at 12 29 19" src="https://user-images.githubusercontent.com/17739465/102010523-de31bb80-3d3e-11eb-9030-5a59e99d4eea.png">
<img width="664" alt="Screenshot 2020-12-13 at 12 28 57" src="https://user-images.githubusercontent.com/17739465/102010524-deca5200-3d3e-11eb-946d-630f6379e1da.png">
### Stakeholders
<!-- @ tag stakeholders of this bug -->
@SouthGoingZax @tfmorris @cdrini @hornc

SouthGoingZax reviewed (changes requested):
This is what I had proposed, but Mek gave this feedback:  https://github.com/internetarchive/openlibrary/issues/3421#issuecomment-734062155

Do you want to try modifying the ```if``` statement to accommodate his feedback?  And then try modifying the message below your change (the one about not storing structured dates) to give guidance on what people should do if they see the date field (just delete its contents?  Try to move it into one of the other date fields?)?

seabelis commented:
I've changed the condition to display the date field if populated (tested successfully populated and unpopulated on my local version). I've added a tip that asks editor to help by removing the content in the date field and populating the birth/death date fields.

mekarpeles commented:
lgtm, ty!
